### PR TITLE
InListSemver matcher

### PR DESCRIPTION
**Pull request by gthea**
# iOS SDK

## What did you accomplish?

- Added `InListSemverMatcher` implementation.

### CODE REVIEW

File: Split/Matchers/Semver/InListSemverMatcher.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    var targetList: Set<Semver> = Set()
+
+    init(data: [String]?,
+         negate: Bool? = nil,
+         attribute: String? = nil,
+         type: MatcherType? = nil) {
+        super.init(negate: negate, attribute: attribute, type: type)
+        self.targetList = Set<Semver>(data?.compactMap { item in
+            Semver.build(version: item)
```
Comment (sanzmauro) [inline Split/Matchers/Semver/InListSemverMatcher.swift:20]: what happen if `item` is a wrong semver version? are you adding a `null` item into the list?

Comment (gthea) [reply Split/Matchers/Semver/InListSemverMatcher.swift:20]: No, `compactMap` will filter out the `nil` values

File: Split/Matchers/Semver/Semver.swift [hunk 2]
```diff
@@ -29,7 +29,15 @@ class Semver: Equatable {
...
-        return try? Semver(version)
+        do {
+            return try Semver(version)
+        } catch SemverParseError.invalidVersionFormat(let message) {
+            Logger.e("\(message)")
+            return nil
+        } catch {
+            Logger.e("An unknown error occurred parsing version")
+            return nil
+        }
```
Comment (javrudsky) [inline Split/Matchers/Semver/Semver.swift:40]: This two returns nil could be replaced by one at the end of the function.

**Review by javrudsky — APPROVED** [review]
👍

**Pull request merged**